x-pack/filebeat/tests/integration/input_reload_test.go

```diff
@@ -0,0 +1,269 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build integration
+
+package integration
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/types/known/structpb"
+
+	"github.com/elastic/beats/v7/x-pack/libbeat/management"
+	"github.com/elastic/elastic-agent-client/v7/pkg/client/mock"
+	"github.com/elastic/elastic-agent-client/v7/pkg/proto"
+)
+
+// TestInputReloadUnderElasticAgent will start a Filebeat and cause the input
+// reload issue described on https://github.com/elastic/beats/issues/33653
+// to happen, then it will check to logs to ensure the fix is working.
+//
+// In case of a test failure the directory with Filebeat logs and
+// all other supporting files will be kept on build/integration-tests.
+//
+// Run the tests wit -v flag to print the temporary folder used.
+func TestInputReloadUnderElasticAgent(t *testing.T) {
+	// We create our own temp dir so the files can be persisted
+	// in case the test fails. This will help debugging issues on CI
+	//
+	// testFailed will be set to 'false' as the very last thing on this test,
+	// it allows us to use t.CleanUp to remove the temporary files
+	testFailed := true
```

Comment:
[NIT]
You could use `testSucceeded`, it'd be more straight forward.

Reply:
It makes sense, thanks!